src/composable/chat/useChatHighlights.ts

```diff
@@ -168,8 +169,13 @@ export function useChatHighlights(ctx: ChannelContext) {
 			ok = h.caseSensitive
 				? msg.body.includes(h.pattern)
 				: msg.body.toLowerCase().includes(h.pattern.toLowerCase());
-		} else if (typeof h.test === "function") {
-			ok = h.test(msg);
+		} else {
+			justTest = true;
+		}
+
+		if (typeof h.test === "function") {
+			const t = h.test(msg);
+			ok = justTest ? t : ok && t;
```

Comment:
What is the purpose of this?

Reply:
When defining a `chatHighligh`, if `regexp`/`pattern` are defined, you can still define `test`, both must pass to return `true` for `checkMatch`.

Originally if both were defined, `test` would be ignored due to the `else if`

Reply:
That doesn't seem useful, there is no reason I can think of where it would be desirable to have both a pattern and a test function. The latter is capable of doing the former anyway.

Reply:
Changes made